alerting/src/main/kotlin/org/opensearch/alerting/util/destinationmigration/DestinationMigrationUtilService.kt

```diff
@@ -64,17 +64,25 @@ class DestinationMigrationUtilService {
             try {
                 runningLock = true
 
-                val destinationsToMigrate = retrieveDestinationsToMigrate(client)
-                logger.info("Need to migrate ${destinationsToMigrate.size} destinations")
-                if (destinationsToMigrate.isEmpty()) {
+                val emailAccountsToMigrate = retrieveConfigsToMigrate(client, "email_account")
+                val emailGroupsToMigrate = retrieveConfigsToMigrate(client, "email_group")
+                val destinationsToMigrate = retrieveConfigsToMigrate(client, "destination")
+                val configsToMigrate = emailAccountsToMigrate + emailGroupsToMigrate + destinationsToMigrate
+                logger.info(
+                    "Need to migrate ${emailAccountsToMigrate.size} email accounts, " +
+                        "${emailGroupsToMigrate.size} email groups and " +
+                        "${destinationsToMigrate.size} destinations " +
+                        "(${configsToMigrate.size} configs total)"
+                )
+                if (configsToMigrate.isEmpty()) {
                     finishFlag = true
```

Comment:
Do we persist this flag? Is this job supposed to run every time the process restarts even if the migration has completed? Wanted to understand the expected lifecycle of this job. It would have been easier if there was a comment on lifecycle!

Reply:
Yeah so the job will finish and deschedule if there are no more Destination configs to migrate.

On a cluster change event, it would reschedule and check if there are any Destination configs (assuming the user didn't restore from snapshot or something this should change since they can't create new ones anymore) and will disable itself again.